execute/plugin.go

```diff
@@ -406,6 +407,7 @@ func (p *Plugin) Outcome(
 				Observation: decodedAos[i].Observation.Contracts,
 			}
 		}
+		p.lggr.Infow("Processing discovery observations", "discoveryObservations", discoveryAos)
```

Comment:
Let's generalize this. Move it to the top of `Outcome(`
```
p.lggr.Debugw("execute Outcome", "outctx": outctx, "query": query", "aos": aos)
```

Reply:
Before decoding would be more general, but require a tool for interpreting data. After would be better for ad-hoc log inspection.

Reply:
Before decoding I believe it would log the base64-encoded bytes, which should be decodable into JSON after getting the raw bytes.